### ISSUE TITLE
Incorrect behavior when indexing with single element

### ISSUE THREAD

**Issue by umar456** [arrayfire/arrayfire#697]
Indexing with single element on a ND array should return the element at the linear location of the array.

```
array A = randu(10, 10);
af_print(A(0)); // Should return one element
```

The above code returns the first row instead of the first element.

**Comment by shehzan10** [comment]
It's because the default arguments for array proxy are span now. They were 0 before proxy was added.

**Comment by umar456** [comment]
They were separate functions before. It should be a simple fix.

**Comment by pavanky** [comment]
The indexing part is fixed. The assignment parts need to be fixed.

**Comment by pavanky** [comment]
Fixed by #720
